NfsDiagnostics/nfsclientlogs.sh

```diff
@@ -85,6 +115,32 @@ start_trace() {
   trace-cmd start -e nfs
 }
 
+dump_os_information() {
+  echo "======= Distro details =======" > os_details.txt
+  cat /etc/os-release >> os_details.txt
+  echo -e "\nKernel version: `uname -a`" >> os_details.txt
+  echo -e "\nNFS Kernel Module information:" >> os_details.txt
+  modinfo nfs >> os_details.txt
+  echo -e "\nLast reboot:" >> os_details.txt
+  last reboot -5 >> os_details.txt
+  echo -e "\nSystem Uptime:" >> os_details.txt
+  cat /proc/uptime >> os_details.txt
+}
+
+dump_debug_stats() {
+  echo -e "\nDate: `date -u`" >> nfs_diag.txt
+  echo -e "\n======= NFS mount stats from mountstats =======" >> nfs_diag.txt
+  mountstats  >> nfs_diag.txt
+  echo -e "\n======= NFS stats from nfsstat  =======" >> nfs_diag.txt
+  nfsstat >> nfs_diag.txt
+  echo -e "\n======= NFS Mounts =======" >> nfs_diag.txt
+  mount -t nfs4 >> nfs_diag.txt
+  echo -e "\n======= NFS TCP Connections =======" >> nfs_diag.txt
+  ss -t | grep nfs >> nfs_diag.txt
+  echo -e "\n======= List of processes in system =======" >> nfs_diag.txt
+  ps -ef >> nfs_diag.txt
```

Comment:
Maybe also dump /proc/*/stack? Even for SmbDiagnostics

Reply:
Ack Will test this out and do in next iteration.